src/backend/common/data/commands/mkcv1importer.py

```diff
@@ -60,3 +65,608 @@ async def handle(self, db_wrapper, s3_wrapper):
+        if not bcrypt.checkpw(self.password.encode('utf-8'), v1_user.password_hash.encode('utf-8')):
+            raise Problem("Invalid login details", status=401)
+        return v1_user
+
+@save_to_command_log
+@dataclass
+class TransferMKCV1UserCommand(Command[UserLoginData]):
+    email: str
+    password_hash: str
+    join_date: int
+    player_id: int | None
+    about_me: str | None
+    user_roles: list[NewMKCUserRole]
+    series_roles: list[NewMKCSeriesRole]
+    team_roles: list[NewMKCTeamRole]
+
+    async def handle(self, db_wrapper, s3_wrapper):
+        async with db_wrapper.connect() as db:
+            row = await db.execute_insert("INSERT INTO users(email, password_hash, join_date, player_id) VALUES (?, ?, ?, ?)", 
+                                          (self.email, self.password_hash, self.join_date, self.player_id))
+            # TODO: Run queries to identify why user creation failed
+            if row is None:
+                raise Problem("Failed to create user")
+            user_id = int(row[0])
+
+            is_banned = False
+            expiration_date: int | None = None
+            if self.player_id:
+                async with db.execute("SELECT expiration_date FROM player_bans WHERE player_id = ?", (self.player_id,)) as cursor:
+                    row = await cursor.fetchone()
+                    if row:
+                        is_banned = True
+                        if row[0]:
+                            expiration_date = int(row[0])
+
+            # add about me from old site data
+            await db.execute("INSERT INTO user_settings(user_id, about_me) VALUES(?, ?)", (user_id, self.about_me))
+
+            # add user, series, team roles
+            insert_user_roles: list[tuple[int, int, int | None]] = [(user_id, roles.id_by_default_role[role.role_name], None) for role in self.user_roles]
+            if is_banned:
+                insert_user_roles.append((user_id, roles.id_by_default_role[roles.BANNED], expiration_date))
+            insert_series_roles = [(user_id, series_roles.id_by_default_role[role.role_name], role.series_id) for role in self.series_roles]
+            insert_team_roles = [(user_id, team_roles.id_by_default_role[role.role_name], role.team_id) for role in self.team_roles]
+            await db.executemany("""INSERT INTO user_roles(user_id, role_id, expires_on) VALUES(?, ?, ?)""", insert_user_roles)
+            await db.executemany("""INSERT INTO user_series_roles(user_id, role_id, series_id) VALUES(?, ?, ?)""", insert_series_roles)
+            await db.executemany("""INSERT INTO user_team_roles(user_id, role_id, team_id) VALUES(?, ?, ?)""", insert_team_roles)
+            await db.commit()
+            return UserLoginData(user_id, self.player_id, self.email, self.password_hash)
+
+@save_to_command_log
+@dataclass
+class ConvertMKCV1DataCommand(Command[None]):
+    async def handle(self, db_wrapper, s3_wrapper) -> None:
+        # TODO (when added to data imports):
```

Comment:
I think we should break this function down into smaller parts, it is too long to be able to understand what it is doing. They can just be functions defined on the `ConvertMKCV1DataCommand` class.